doc/extensions/README.md

```diff
@@ -0,0 +1,33 @@
+Extension Documentation
+=======================
+
+
+What is an Extension?
+---------------------
+
+> An extension adds/modifies/removes commands that are part of `az` and can be optionally installed by the user.
+
+- Extensions are supported through all installation methods.
+- Extensions are the way to go to release private/public previews of your Azure CLI commands.
+- Currently, we support one extension type, a [Python Wheel](http://pythonwheels.com/).
+- All extension documentation here refers to this type of extension.
+
+
+What an Extension is not
+------------------------
+
+- An extension is not an external executable or product outside of `az`.
+
+- If you're looking to create a command module for `az`, see [authoring command modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules).
```

Comment:
I assume here it means to create a command module which is installed by default? If yes, I suggest we call it out otherwise it look rather similar with an extension module